packages/medusa/src/api/routes/admin/products/get-product.ts

```diff
@@ -136,7 +137,9 @@ export default async (req, res) => {
         },
       }
       salesChannels = await remoteQuery(query)
-    } else {
+    } else if (
+      featureFlagRouter.isFeatureEnabled(SalesChannelFeatureFlag.key)
```

Comment:
Only in that specific case we can continue and use the `salesChannelService` to add new relations to the Product.

Reply:
You right, I added a new commit to raise the condition at a higher level